sonic_sku_create/sonic_sku_create.py

```diff
@@ -0,0 +1,329 @@
+	
+	def print_port_config(self) :
+		#print a port_config.ini file based on the sku definition 
+		
+		header = PORTCONFIG_HEADER # ["name", "lanes", "alias", "index"]
+		port_config = []
+		for line in self.portconfig_dict.values():
+			port_config.append(line)
+		 
+		port_config.sort(key=lambda x: (int(re.search(('\d+'),x[0]).group(0)))) # sort the list with interface name
+		print(tabulate(port_config, header,tablefmt="plain"))
+		
+		
+	def create_sku_dir(self) :
+		# create a new SKU directory based on the base SKU
+		try:
+			shutil.copytree(self.base_sku_dir, self.new_sku_dir)
+		except OSError as e:
+			print(e.message, file=sys.stderr) 
+	
+	def remove_sku_dir(self) :
+		# remove SKU directory 
+		try:
+			if (self.base_sku_dir == self.new_sku_dir) :
+				print ("Removing the base SKU" + self.new_sku_dir + " is not allowed")
+				return
+			shutil.rmtree(self.new_sku_dir)
+		except OSError as e:
+			print(e.message, file=sys.stderr) 
+					
+	def platform_specific(self) :
+		func = self.platform_specific_dict.get(self.platform, lambda: "nothing")
+		return func()
+		
+		
+	def msn2700_specific(self) :
+		for fp, values in self.fpp_split.items():
+			splt_arr = sorted(values[0])
+			idx_arr = sorted(values[1])
+			
+			splt = len(splt_arr)
+			fp=int(fp)
+			if ((fp%2) == 1 and splt == 4) :
+				try :
+					self.portconfig_dict.pop(str(int(idx_arr[-1])+1) )
+				except KeyError:
+					continue
+				print ("MSN2700 -  Disabled front panel port ",fp+1, "due to port ",fp,"Split by 4") 
+			elif ((fp%2) == 0 and splt == 4) :
+				for idx in idx_arr :
+					self.portconfig_dict.pop(idx,None )
+				print ("MSN2700 -  even front panel ports (",fp,") are not allowed to split by 4")
+				return -1
+
+
+	def l2_mode(self,l2_sku) :
+		if not os.path.exists(DEFAULT_DEV_PATH+self.platform+"/"+l2_sku):
+			print ("Error - path:", l2_sku, " doesn't exist")
+		print( "Starting L2 mode configuration based on "+l2_sku+" SKU")	
+		cfg_db_str = "\n{\n\t\"DEVICE_METADATA\": {\n\t\t\"localhost\": {\n\t\t\t\"hostname\": \"sonic\"\n\t\t}\n\t}\n}\n"
+
+		subprocess.call("cat <<EOF | sudo config reload /dev/stdin -y" + cfg_db_str, shell=True)
+		subprocess.call("sudo sonic-cfggen -H --write-to-db ", shell=True)
+		subprocess.call("sonic-cfggen -t /usr/local/lib/python2.7/dist-packages/usr/share/sonic/templates/l2switch.j2 -p -k" + l2_sku + " | sudo config load /dev/stdin -y", shell=True ) 
+		subprocess.call("sudo config save -y ", shell=True)
+		subprocess.call("sudo systemctl restart swss ", shell=True)
+
+		print( "Finished L2 mode configuration based on "+l2_sku+" SKU")
+		
+def main():
+	parser = argparse.ArgumentParser(description='Create a new SKU',
+									version='1.0.0',
+									formatter_class=argparse.RawTextHelpFormatter)
+	parser.add_argument('-f', '--file', action='store', help='SKU definition file (default <cwd>/sku_def.xml)', default=None)
+	parser.add_argument('-b', '--base', action='store', help='SKU base definition  ', default=None)
```

Comment:
>None [](start = 91, length = 4)

Actually it's easy to find the base SKU inside a running SONiC, so we could safely have a default value.

```
PLATFORM=`sonic-cfggen -H -v DEVICE_METADATA.localhost.platform`
PRESET=(`head -n 1 /usr/share/sonic/device/$PLATFORM/default_sku`)
echo ${PRESET[0]}
```

Reply:
Changed to using "sonic-cfggen -H -v DEVICE_METADATA.localhost.platform"

Reply:
I mean the default value of 'base' is known from the image itself, and we don't need user feed this argument on command line.

---
In reply to: [293090385](https://github.com/Azure/sonic-utilities/pull/547#discussion_r293090385) [](ancestors = 293090385)